plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -0,0 +1,276 @@
+        }
+    }
+
+    private static Map<String, String> icebergTableProperties(String location, Map<String, String> hiveTableProperties, String schemaNameMapping)
+    {
+        Map<String, String> icebergTableProperties = new HashMap<>();
+
+        icebergTableProperties.putAll(hiveTableProperties);
+        EXCLUDED_PROPERTIES.forEach(icebergTableProperties::remove);
+
+        icebergTableProperties.put("provider", "iceberg");
+        icebergTableProperties.put("migrated", "true");
+        icebergTableProperties.putIfAbsent("location", location);
+        icebergTableProperties.put(DEFAULT_NAME_MAPPING, schemaNameMapping);
+        icebergTableProperties.put(FORMAT_VERSION, String.valueOf(FORMAT_VERSION_SUPPORT_MAX));
+
+        return ImmutableMap.copyOf(icebergTableProperties);
+    }
+
+    private Schema toIcebergSchema(List<Column> columns)
+    {
+        List<Types.NestedField> icebergColumns = new ArrayList<>();
+        for (Column column : columns) {
+            int index = icebergColumns.size();
+            org.apache.iceberg.types.Type type = toIcebergType(typeManager.getType(column.getType().getTypeSignature()));
+            Types.NestedField field = Types.NestedField.of(index, false, column.getName(), type, column.getComment().orElse(null));
+            icebergColumns.add(field);
+        }
+        org.apache.iceberg.types.Type icebergSchema = Types.StructType.of(icebergColumns);
+        AtomicInteger nextFieldId = new AtomicInteger(1);
+        icebergSchema = TypeUtil.assignFreshIds(icebergSchema, nextFieldId::getAndIncrement);
+        return new Schema(icebergSchema.asStructType().fields());
+    }
+
+    private List<DataFile> buildDataFiles(ConnectorSession session, HiveStorageFormat format, String location, PartitionSpec partitionSpec, String partitionKey)
+            throws IOException
+    {
+        Path locationPath = new Path(location);
+        FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsEnvironment.HdfsContext(session), locationPath);
+        List<FileStatus> fileStatus = Arrays.stream(fileSystem.listStatus(locationPath, HIDDEN_PATH_FILTER))
+                .filter(FileStatus::isFile)
+                .collect(toImmutableList());
+        DataFile[] datafiles = new DataFile[fileStatus.size()];
+
+        Tasks.range(fileStatus.size())
+                .stopOnFailure()
+                .throwFailureWhenFinished()
+                .run(index -> datafiles[index] = buildDataFile(fileStatus.get(index), partitionKey, partitionSpec, format.name()));
+
+        return Arrays.stream(datafiles).collect(toImmutableList());
+    }
+
+    private static List<String> getPartitionColumnNames(io.trino.plugin.hive.metastore.Table table)
+    {
+        return table.getPartitionColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+    }
+
+    private static String toSchemaNameMapping(Schema schema)
+    {
+        return schema.columns().stream()
```

Comment:
Maybe better to construct a NameMapping instance and then use the NameMappingParser to convert to JSON?